### PR TITLE
refactor(distributions): standardize output types

### LINKED ISSUE #88: FIX: Handle scalar case in tests
There is some issues in mathematical functions in distributions and mixture with scalar and non-scalar inputs. For example in `core/mixture.py` on lines 234-253:

```python
    def pdf(self, X: ArrayLike) -> NDArray[float64]:
        """Probability Density Function of the mixture.

        The PDF is computed as the weighted sum of the PDFs of its
        components.

        Parameters
        ----------
        X : ArrayLike
            The input data points at which to evaluate the PDF.

        Returns
        -------
        NDArray[np.float64]
            The PDF values corresponding to each point in :attr:`X`.
        """

        X = np.asarray(X, dtype=float64)
        component_pdfs = np.array([comp.pdf(X) for comp in self.components])
        return np.asarray(np.dot(self.weights, component_pdfs))

```

Make sure that all functions returns scalar if the input is scalar and return ndarray with internal dtype if the input is non-scalar.
Add tests to it.

### LINKED ISSUE #93: REFACTOR: Standardize output types in distributions
It's necessary for distribution methods to return a scalar when given a scalar, and to return an array when given an array.
It is also necessary to test both cases.

There's also a problem: in some distributions, the pdf method is calculated using a formula rather than by exponentiating the lpdf result, which can degrade numerical stability. A decision needs to be made.

### PR DESCRIPTION
**Pull request by xImoZA**
This PR standardizes the behavior of distributions to ensure consistent return types (scalars vs arrays) and improves the numerical stability of PDF calculations.

## Changes
- **Numerical Stability:** Refactored `pdf` methods in all distributions to calculate probability density via `np.exp(lpdf(X))`. 
- **Standardized Outputs:** Updated distribution methods (`pdf`, `lpdf`, `ppf`, `log_gradients`) to strictly distinguish between inputs:
  - Scalar input -> Scalar output.
  - Array input -> Array output with correct `dtype`.
- **Generate:** - Update `generate` method signature in `ContinuousDistribution` and implementations to support `size` as `int`, `tuple`, or `None` (SciPy-like behavior).
- **Testing:** Added comprehensive property-based tests (via Hypothesis) for all distributions ensuring that:
  - Scalar inputs return scalars (`np.isscalar` check).
  - Array inputs return arrays with correct shapes and dtypes.
  - Add unit tests covering various `size` shapes (scalar, 1D, ND, empty)

### CODE REVIEW

**Comment by VITYANA** [comment]
good code :shipit:

File: rework_pysatl_mpest/core/mixture.py [hunk 1]
```diff
@@ -268,15 +268,16 @@ def pdf(self, X: ArrayLike) -> NDArray[DType]:
...
-        NDArray[DType]
+        DType | NDArray[DType]
             The PDF values corresponding to each point in :attr:`X`.
+            Return a scalar when given a scalar, and to return an array when given an array.
         """
 
         X = np.asarray(X, dtype=self.dtype)
         component_pdfs = np.array([comp.pdf(X) for comp in self.components])
-        return np.asarray(np.dot(self.weights, component_pdfs))
+        return np.dot(self.weights, component_pdfs)
```
Comment (iraedeus) [inline rework_pysatl_mpest/core/mixture.py]: Why not just `return np.exp(self.lpdf(X))`?

File: rework_pysatl_mpest/distributions/beta.py [hunk 2]
```diff
@@ -189,7 +199,7 @@ def lpdf(self, X):
         log_pdf_standard = beta_dist.logpdf(Z, self.alpha, self.beta).astype(dtype)
         result = log_pdf_standard - np.log(self.right_border - self.left_border)
 
-        return np.atleast_1d(result)
+        return result
```
Comment (iraedeus) [inline rework_pysatl_mpest/distributions/beta.py:205]: Here, the result is a 0-dimensional array if a scalar is passed to the method. This differs from other distributions (in particular, from the behavior of mixture).

**Review by iraedeus — CHANGES_REQUESTED** [review]
1. In some places a scalar is returned, and in others a zero-dimensional array is returned. I suggest returning a scalar everywhere (using `result[()]`)
2. I also propose standardizing the 'generate' method, making it scipy/numpy-like. To do this, simply pass `size` to the scipy generator. Size parameter should be of type `int | tuple[int, ...] | None`

**Review by iraedeus — APPROVED** [review]
